host/signalworkflow_test.go

```diff
@@ -586,6 +586,121 @@ CheckHistoryLoopForSignalSent:
 	s.Equal("history-service", *signalEvent.WorkflowExecutionSignaledEventAttributes.Identity)
 }
 
+func (s *integrationSuite) TestSignalWorkflow_Cron_NoDecisionTaskCreated() {
+	id := "integration-signal-workflow-test-cron"
+	wt := "integration-signal-workflow-test-cron-type"
+	tl := "integration-signal-workflow-test-cron-tasklist"
+	identity := "worker1"
+	cronSpec := "@every 2s"
+	activityName := "activity_type1"
+
+	workflowType := &workflow.WorkflowType{}
+	workflowType.Name = common.StringPtr(wt)
+
+	taskList := &workflow.TaskList{}
+	taskList.Name = common.StringPtr(tl)
+
+	// Start workflow execution
+	request := &workflow.StartWorkflowExecutionRequest{
+		RequestId:                           common.StringPtr(uuid.New()),
+		Domain:                              common.StringPtr(s.domainName),
+		WorkflowId:                          common.StringPtr(id),
+		WorkflowType:                        workflowType,
+		TaskList:                            taskList,
+		Input:                               nil,
+		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(100),
+		TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(1),
+		Identity:                            common.StringPtr(identity),
+		CronSchedule:                        &cronSpec,
+	}
+	now := time.Now()
+
+	we, err0 := s.engine.StartWorkflowExecution(createContext(), request)
+	s.Nil(err0)
+
+	s.Logger.Info("StartWorkflowExecution", tag.WorkflowRunID(*we.RunId))
+
+	// Send first signal using RunID
+	signalName := "my signal"
+	signalInput := []byte("my signal input.")
+	err := s.engine.SignalWorkflowExecution(createContext(), &workflow.SignalWorkflowExecutionRequest{
+		Domain: common.StringPtr(s.domainName),
+		WorkflowExecution: &workflow.WorkflowExecution{
+			WorkflowId: common.StringPtr(id),
+			RunId:      common.StringPtr(*we.RunId),
+		},
+		SignalName: common.StringPtr(signalName),
+		Input:      signalInput,
+		Identity:   common.StringPtr(identity),
+	})
+	s.Nil(err)
+
+	// decider logic
+	activityScheduled := false
+	activityData := int32(1)
+	var decisionTaskDelay time.Duration
+	dtHandler := func(execution *workflow.WorkflowExecution, wt *workflow.WorkflowType,
+		previousStartedEventID, startedEventID int64, history *workflow.History) ([]byte, []*workflow.Decision, error) {
+		decisionTaskDelay = time.Now().Sub(now)
+		if !activityScheduled {
```

Comment:
I don't think you need anything related to activity. Just complete workflow execution directly in the dtHandler should be enough.